api/src/controllers/record-utils.js

```diff
@@ -240,17 +219,18 @@ const createRecordByJSON = data => {
     throw new PublicError('Form not found: ' + options.form);
   }
 
-  const formData = {};
-
-  // For now only save string and number fields, ignore the others.
-  // Lowercase all property names.
-  for (var k in data) {
-    if (['string', 'number'].indexOf(typeof data[k]) >= 0) {
-      formData[k.toLowerCase()] = data[k];
+  const correctedData = {};
+  Object.keys(data).forEach(k => {
+    if (formDefinition.fields[k]  && formDefinition.fields[k].type === 'date') {
+      // Of the data types we support, date is the only one that can't be natively supported in
+      // JSON. Convert it the same way smsparser does, into ms ts
+      data[k] = moment(data[k]).valueOf();
```

Comment:
this is the only field type we say we support that JSON doesn't natively support so we need to convert it